text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+Patch and minor updates may happen at any time.
+
+Major updates will happen no more than once per year unless necessitated by a significant security concern.
+
+### General Support Policy
+
+Each major release of an ESL crate will receive backwards-compatible updates, at least addressing security and stability issues, for two years after the release of the subsequent major release. For example - if version 2.x.x is released on January 1, 2027, then version 1.x.x will receive updates until January 1, 2029.
+
+The only exception to this policy is if resolving a security issue requires a new major release (i.e., cannot be fixed in a backwards-compatible manner). If this happens, all previous vulnerable major versions will cease to receive updates immediately.
+
+### Security Updates
+
+The ESL security policy should largely mirror the [Go Security Policy](https://go.dev/doc/security/policy).
+
+Reports will be acknowledged within 7 days, and each vulnerability will be assigned a track - PUBLIC, PRIVATE, or URGENT. All security issues will be issued CVE numbers.
+
+Security updates will be created and released as quickly as reasonably possible.
+
+URGENT track issues are fixed in private and trigger an immediate dedicated security release, possibly with no pre-announcement.
+
+### Code Review and Testing Standards
+
+Rust ESL code will be managed in GitHub repositories. GitHub will be used for managing issues and pull requests.
+
+The Rust ESL project will maintain a list of maintainers who are qualified to review PRs.
+
+Any change to ESL code must be reviewed by a maintainer who is not the patch author. This will be enforced by GitHub settings. Pull requests will be required, no direct merges will be allowed.
+
+Whenever possible, changes to functionality should come with tests. When it is reasonably possible to write a test, no changes to functionality will be merged until they are merged with tests.
+
+Rust code in the ESL will follow the [Rust Style Guide](https://github.com/rust-lang/rust/tree/HEAD/src/doc/style-guide/src).
+
+API styles for ESL crates should become consistent over time. At minimum, it should be a goal to converge on the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/about.html). This does not need to happen prior to a crate joining the ESL, but ESL APIs should evolve towards this ideal at an acceptable rate of change.
+
+### Minimum Supported Rust Version (MSRV) Guarantees
+
+A maximum MSRV will be selected for the entire ESL and updated as it makes sense to do so. The maximum MSRV will never be newer than two years old. Individual crates may have MSRVs older than the maximum.
+
+### Platform Support
+
+Platform support will be managed in the same way that [platform support is managed by the Rust language](https://doc.rust-lang.org/nightly/rustc/platform-support.html), with Tier 1 platform guaranteed to work, Tier 2 platforms guaranteed to build, and Tier 3 platforms with non-guaranteed support.
```

Comment:
There should be an exception here for platforms where the functionality isn't needed or can't be implemented. We might also want exceptions where it would take a large amount of (unfunded) effort to make a crate work.

How would the relationship between Rust tier approvals and the ESL work?
Would approvals be gated on the ESL, or would the ESL have to adapt to new approvals?